src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -1248,6 +1262,7 @@ else if (i == 0) {
       }
       if (schemaPattern == null) {
         schemaPattern = session.getSchema();
+        useSessionSchema = true;
```

Comment:
Set the value useSessionSchema to true whenever the schema value is being used from the connection context.